### PR TITLE
Add existing_connection function to ConnectionCache

### PR DESCRIPTION
**Pull request by acco**
This pull request introduces several improvements to the connection handling and database operations:

1. Added a new `existing_connection/2` function to `ConnectionCache` to retrieve only existing connections without creating new ones.

2. Modified `find_or_create_connection/3` to accept a `create_on_miss` parameter, allowing more control over connection creation.

3. Updated `with_connection/2` in the `Databases` module to use `existing_connection/2` first, falling back to creating a temporary connection if needed.

4. Removed the timeout logic from `with_connection/2`, simplifying the function.

5. Added a new test case for `ConnectionCache` to verify error handling when a connection is not found and `create_on_miss` is false.

6. Updated factory functions in `CharacterFactory` to use more consistent and deterministic data generation methods.

7. Added `changeset/2` functions to the Character-related models, improving data validation and insertion.

8. Modified the `postgres_database_attrs/1` function in `DatabasesFactory` to set very low queue target and interval values for testing purposes.

These changes improve the flexibility and reliability of database connection handling while also enhancing the test suite and data generation processes.

### CODE REVIEW

**Comment by acco** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/123?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#123** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/123?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#122** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/122?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#121** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/121?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#120** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/120?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#119** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/119?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#118** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/118?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#117** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/117?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#116** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/116?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#115** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/115?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#114** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/114?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#113** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/113?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#112** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/112?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#111** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/111?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#110** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/110?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @acco and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: lib/sequin/databases/connection_cache.ex [hunk 1]
```diff
@@ -114,24 +115,26 @@ defmodule Sequin.Databases.ConnectionCache do
       }
     end
 
-    @spec find_or_create_connection(t(), database()) :: {:ok, pid(), t()} | {:error, term()}
-    def find_or_create_connection(%__MODULE__{} = state, db) do
+    @spec find_or_create_connection(t(), database(), boolean()) :: {:ok, pid(), t()} | {:error, term()}
+    def find_or_create_connection(%__MODULE__{} = state, db, create_on_miss) do
```
Comment (RTLS) [inline lib/sequin/databases/connection_cache.ex:119]: This leaves me wondering why we have `connection` functions both in ConnectionCache and in Databases.

My read is that `Databases.with_connection/1` is a convenience utility. Could we instead move that into `ConnectionCache`?

Not sure that there is a significant cost to having connection access in a few different modules but might be a smell

File: lib/sequin/databases/connection_cache.ex [hunk 2]
```diff
@@ -168,7 +171,12 @@ defmodule Sequin.Databases.ConnectionCache do
...
   @spec connection(GenServer.server(), database()) :: start_result()
   def connection(server \\ __MODULE__, %PostgresDatabase{} = db) do
-    GenServer.call(server, {:connection, db})
+    GenServer.call(server, {:connection, db, true})
+  end
+
+  @spec existing_connection(GenServer.server(), database()) :: start_result() | {:error, NotFoundError.t()}
+  def existing_connection(server \\ __MODULE__, %PostgresDatabase{} = db) do
+    GenServer.call(server, {:connection, db, false})
   end
```
Comment (RTLS) [inline lib/sequin/databases/connection_cache.ex:180]: these are great 👍

File: lib/sequin/databases/connection_cache.ex (hunk 1 above)
Comment (RTLS) [reply lib/sequin/databases/connection_cache.ex:119]: I guess `ConnectionCache` is responsible simply for starting and keeping track of connections to a database.

`Databases.with_connection` utilizes that to actually run a query. That's a good separation actually. So I'm on board with these APIs.

**Review by RTLS — APPROVED** [review]

**Pull request merged**